libraries/c_sdk/standard/mqtt/test/unit/iot_tests_mqtt_api.c

```diff
@@ -503,6 +511,140 @@ static void _decrementReferencesJob( IotTaskPool_t pTaskPool,
     }
 }
 
+/**
+ * @brief Transport send interface provided to the MQTT context used in calling MQTT LTS APIs.
+ */
+static int32_t transportSend( NetworkContext_t * pNetworkContext,
+                              const void * pMessage,
+                              size_t bytesToSend )
+{
+    int32_t bytesSend = 0;
+
+    IotMqtt_Assert( pNetworkContext != NULL );
+    IotMqtt_Assert( pMessage != NULL );
+
+    /* Sending the bytes on the network using Network Interface. */
+    bytesSend = pNetworkContext->pNetworkInterface->send( pNetworkContext->pNetworkConnection, ( const uint8_t * ) pMessage, bytesToSend );
+
+    if( bytesSend < 0 )
```

Comment:
Do we need to handle the case where 0 < bytesSend < bytesToSend?  Maybe a loop?  Or is there a guarantee that the lower layer will return an error instead?  Is there a demo and does it behave the same?